### ISSUE TITLE
[apex] AvoidNonExistentAnnotations rule's line/col info should be the annotation - not the entire class or method

### ISSUE THREAD

**Issue by stephen-carter-at-sf** [pmd/pmd#5613]
**Affects PMD Version:** 7.11.0

Make sure, to test with the latest PMD version.

**Description:**

When running PMD against the example given in https://docs.pmd-code.org/latest/pmd_rules_apex_errorprone.html#avoidnonexistentannotations
I see that the range (start and end line and column numbers) are for the entire class instead of the annotation itself. This makes for a super noisy when integrated with other tools (like the Salesforce Code Analyzer VS Code Extension).

I see the other tools like vscode-apex-pmd fix this issue (see https://github.com/ChuckJonas/vscode-apex-pmd/blob/master/src/lib/apexPmd.ts#L77-L79) by only ever highlighting the first line of a violation.

As the lead developer of the Salesforce Code Analyzer, I'd prefer to not have to modify the violation range but use the one provided by PMD itself.

**Exception Stacktrace:**

```
# Copy-paste the stack trace here
```

**Code Sample demonstrating the issue:**

```java

```

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

1.  ... (e.g. if you're using maven: `mvn clean verify`)
2.  ...

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
